src/nvim/buffer.c

```diff
@@ -279,26 +280,6 @@ bool buf_valid(buf_T *buf)
   return false;
 }
 
-/// A hash table used to quickly lookup a buffer by its number.
-static hashtab_T buf_hashtab;
-
-static void buf_hashtab_add(buf_T *buf)
-{
-  snprintf((char *)buf->b_key, sizeof(buf->b_key), "%x", buf->b_fnum);
-  if (hash_add(&buf_hashtab, buf->b_key) == FAIL) {
-    EMSG(_("E931: Buffer cannot be registered"));
-  }
-}
-
-static void buf_hashtab_remove(buf_T *buf)
```

Comment:
Let's keep this and `buf_hashtab_add` as wrappers, so that it's clear if future Vim patches change something.